File: 3000-maximum-area-of-longest-diagonal-rectangle/Solution.js
var areaOfMaxDiagonal = function (dimensions) {
    let longDiag = 0;
    let maxArea = 0;

    dimensions.forEach((rect) => {
        let diag = Math.sqrt((rect[0] * rect[0]) + (rect[1] * rect[1]));
        let area = rect[0] * rect[1];

        if (diag > longDiag) {
            longDiag = diag;
            maxArea = area;
        } else if (diag == longDiag && area > maxArea) {
            maxArea = area;
        }
    });

    return maxArea;
};

File: 3000-maximum-area-of-longest-diagonal-rectangle/Solution.java
class Solution {
    public int areaOfMaxDiagonal(int[][] dim) {
        double longDiag = 0;
        double maxArea = 0;

        for (int i = 0; i < dim.length; i++) {
            double diag = Math.sqrt((dim[i][0] * dim[i][0]) + (dim[i][1] * dim[i][1]));
            double area = dim[i][0] * dim[i][1];

            if (diag > longDiag) {
                longDiag = diag;
                maxArea = area;
            } else if (diag == longDiag && area > maxArea) {
                maxArea = area;
            }
        }

        return (int) maxArea;
    }
}
